extensions/resume.md

```diff
@@ -0,0 +1,255 @@
+
+Upon establishing the new connection, the client begins capability negotiation, negotiates all mutually-supported capabilities, and MUST confirm that the `draft/resume-0.3` capability exists. If this capability does not exist, the client continues connection registration without attempting to resume. If this capability does exist, the client sends the `RESUME` command and MUST wait for either a `RESUME SUCCESS` or a `RESUME ERR` message before continuing registration. It should be noted that the client MUST NOT perform SASL authentication if the `draft/resume-0.3` capability exists, as this will end connection registration and abort the resumption attempt before the resume attempt can complete.
+
+If the token provided by the new client is validated by the server, and both the old and new clients use TLS, then the attempt SHOULD be successful. If the attempt is successful, the server MUST send a `RESUME SUCCESS` message and complete connection registration immediately (at which time the state will begin to replay as described below). If the attempt is unsuccessful, the server MUST send a `RESUME ERR` message, MAY send any additional error numerics, and then allows the client to continue connection registration.
+
+If the client receives a `RESUME SUCCESS` message, connection registration will complete immediately and state will begin to replay as described below. If the client receives a `RESUME ERR`, the client MUST continue registration as though they are joining the server normally.
+
+On a successful request, the server:
+
+1. Terminates the old client's connection.
+2. Updates the client's username, hostname, and resume token (or lack of one) to match the new client's information.
+3. Plays the 'joining server' burst to the new client.
+4. Replays the client's session to the new client.
+5. Sends `RESUMED` or `QUIT`+`JOIN` messages to other clients as appropriate.
+6. Sends clients that have the reconnecting user `MONITOR`'d one `RPL_MONOFFLINE` numeric and one `RPL_MONONLINE` numeric indicating that the user has reconnected. The timestamps on both these messages, if sent, SHOULD be the current time.
+7. If message history could not be replayed (because it is not stored, or for any other reason), sends the new client a `RESUME WARN` message making them aware of this.
+
+On a successful request, the session information that MUST be applied from the old client and replayed includes, but is not limited to:
+
+- Joined channels, along with channel membership prefixes (op, halfop, etc).
+- Metadata.
+- MONITOR list.
+- User modes.
+- Logged-in user account.
+
+Session information that MUST NOT be applied from the old client and replayed includes:
+
+- Enabled client capabilities.
+- The old client's resume token (after completing the resume request, this token is thrown away).
+    - Including both the old client's resume ID and resume key, if using the token values described above.
```

Comment:
This either no longer applies, or is referring to the values described below

Reply:
Oh, yeah I moved things around. wilfix, thanks

Reply:
Fixed this whole line up to be a lot clearer, thanks!